pki.spec

```diff
@@ -135,7 +135,7 @@ ExcludeArch: i686
 %define pki_uid 17
 %define pki_groupname pkiuser
 %define pki_gid 17
-%define pki_homedir /usr/share/pki
+%define pki_homedir /home/%{pki_username}
```

Comment:
pki user is a daemon user which cannot login to the system. I found strange that it has an home directory. Could we have something in `/var` folder instead of `/home`?

Reply:
So there are 2 methods to create SoftHSM tokens, under `/var/lib/softhsm/tokens` or under `$HOME/.config/softhsm2/tokens` (see https://github.com/dogtagpki/pki/wiki/SoftHSM). For the first one the user needs to be added into `ods` group, so it needs root's assistance. The second one doesn't need root's assistance, but the user needs a home directory.

Either way the service will run as an unprivileged user, but the second one will make it possible for people to run a (private) PKI service on machines where they don't have a root access at all. The changes in this PR will make it consistent for all PKI services, i.e. they have a home directory to store private files, and will eliminate the need to modify the `ods` group.

FYI, the current SoftHSM tests are using the first method:
https://github.com/dogtagpki/pki/blob/master/.github/workflows/ca-hsm-test.yml#L55-L69

Reply:
In other words, without home directory `pkiuser` needs a privilege escalation (i.e. adding into `ods` group), which we don't want for a rootless service.

Reply:
If the instance is created and executed by a non root user, is it running as pkiuser or as the user creating/running the instance. I am expecting the second case and that user already has an home folder. I am not getting why the unprivileged  user should execute the instance as pkiuser.

Reply:
Yes, if an instance is created by a non-root user (e.g. `testuser`) that instance will run as that user, and everything that the instance needs is encapsulated in that user's home directory. If an instance is created by root, it will run as `pkiuser` by default. However, without a home directory `pkiuser` will require additional steps (including the privilege escalation which might open access to other tokens as well), additional docs and testing since it's not consistent with other non-root users, and the files will be scattered throughout the system (which needs to be secured properly). If it has a home directory it'll be easier to manage.

Reply:
The examples are for running the service as a non root user and in that scenario we agree to put everything in the home folder. `pkiuser` is a non root user but it is used in a different scenario which could require a different setup.
Even from the design point of view I do not fully agree with this, from the implementation prospective it could simplify the installation code. Therefore, I am accepting for now and after fixing the installation for non root user we can review the process again.